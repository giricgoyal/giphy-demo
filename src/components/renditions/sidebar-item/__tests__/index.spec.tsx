import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import SidebarItem, { Props } from '../index'

describe('<SidebarItem />', () => {
    let container, getByTestId
    const onSelectHandler = jest.fn()

    const defaultProps: Props = {
        item: 'captain_america',
        isActive: false,
        onSelect: onSelectHandler,
    }

    describe('When isActive is false', () => {
        beforeEach(() => {
            ;({ container } = render(<SidebarItem {...defaultProps} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When isActive is true', () => {
        beforeEach(() => {
            const props = {
                ...defaultProps,
                isActive: true,
            }
            ;({ container, getByTestId } = render(<SidebarItem {...props} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })

        describe('onSelect', () => {
            describe('when clicked', () => {
                beforeEach(() => {
                    fireEvent.click(getByTestId('sidebar-item'))
                })

                test('should call the onSelectHandler', () => {
                    expect(onSelectHandler).toHaveBeenCalledWith('captain_america')
                })
            })
        })
    })
})
