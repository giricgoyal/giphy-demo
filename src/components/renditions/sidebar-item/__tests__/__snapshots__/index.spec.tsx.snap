// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItem /> When isActive is false should match snapshot 1`] = `
<div>
  <div
    class="renditions__sidebar__item "
    data-testid="sidebar-item"
  >
    Captain America
  </div>
</div>
`;

exports[`<SidebarItem /> When isActive is true should match snapshot 1`] = `
<div>
  <div
    class="renditions__sidebar__item active"
    data-testid="sidebar-item"
  >
    Captain America
  </div>
</div>
`;
